test/Transactions/Orleans.Transactions.Tests/Runners/GoldenPathTransactionTestRunner.cs

```diff
@@ -14,30 +15,40 @@ protected GoldenPathTransactionTestRunner(IGrainFactory grainFactory, ITestOutpu
 
         [SkippableTheory]
         [InlineData(TransactionTestConstants.SingleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.DoubleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.EightStateTransactionalGrain)]
         public virtual async Task SingleGrainReadTransaction(string transactionTestGrainClassName)
         {
             const int expected = 0;
 
             ITransactionTestGrain grain = RandomTestGrain(transactionTestGrainClassName);
-            int actual = await grain.Get();
-            Assert.Equal(expected, actual);
+            var actualResults = await grain.Get();
+            //each transaction state should all be 0 since no operation was applied yet
+            foreach (var actual in actualResults)
+            {
+                Assert.Equal(expected, actual);
+            }
         }
 
         [SkippableTheory]
         [InlineData(TransactionTestConstants.SingleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.DoubleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.EightStateTransactionalGrain)]
         public virtual async Task SingleGrainWriteTransaction(string transactionTestGrainClassName)
         {
             const int delta = 5;
             ITransactionTestGrain grain = RandomTestGrain(transactionTestGrainClassName);
-            int original = await grain.Get();
+            var original = await grain.Get();
             await grain.Add(delta);
-            int expected = original + delta;
-            int actual = await grain.Get();
+            var expected = original.Select(value => value += delta).ToArray();
+            var actual = await grain.Get();
             Assert.Equal(expected, actual);
         }
 
         [SkippableTheory]
         [InlineData(TransactionTestConstants.SingleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.DoubleStateTransactionalGrain)]
+        [InlineData(TransactionTestConstants.EightStateTransactionalGrain)]
```

Comment:
It's probably ok, but it should be noted that the way these tests are written, transactions will now be tested against TransactionTestConstants.MaxCoordinatedTransactions * TransactionTestConstants.MaxCoordinatedTransactions transactional states.

Reply:
I thought that's the goal?

Reply:
the task is add Golden path multi-resource transactional grain testing

Reply:
Understood, and I'm not necessarily pushing back on this approach, only pointing out that it's a bit beyond what we scoped, and that has implications.  The number of transactional resources is unbounded, but we've committed to testing the system up to 8 for now, which we've defined as TransactionTestConstants.MaxCoordinatedTransactions.  This value will likely change to 16 or 32 once the system is more stable.  The number of transactional resources is the sum of the resource across all the grain, but the original tests, built against grains with a single resource, treated the number of grains and the number of transaction resources as the same, which they are not.  Not correcting this in this pr means we're testing many more resources than we committed to.  If that works without incident, great, but if we encounter issues, we should consider scoping back these tests to the original resource goals.